invenio_files_rest/views.py

```diff
@@ -814,6 +839,49 @@ def delete(self, bucket=None, key=None, version_id=None, upload_id=None,
             obj = self.get_object(bucket, key, version_id)
             return self.delete_object(bucket, obj, version_id)
 
+
+class ObjectVersionTagResource(ContentNegotiatedMethodView):
+    """Tag resource."""
+
+    def __init__(self, *args, **kwargs):
+        """Instantiate content negotiated view."""
+        super(ObjectVersionTagResource, self).__init__(*args, **kwargs)
+
+    @pass_bucket
+    @need_bucket_permission('bucket-update')
+    def put(self, key=None, bucket=None):
+        """Create or update tags."""
+        tags = request.get_json()
+        obj = ObjectVersion.get(bucket, key)
+        if tags is not None:
+            with db.session.begin_nested():
+                for key, value in tags.items():
+                    ObjectVersionTag.create_or_update(obj, key, value)
+            db.session.commit()
+        return self.make_response(
```

Comment:
Seems like a dedicated serializer/deserializer is missing for tags.

Reply:
True, I can add this.